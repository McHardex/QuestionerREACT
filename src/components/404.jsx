import React from 'react';

const NOTFOUND = () => (
  <div>
    <h1>Page not found bro</h1>
  </div>
);

export default NOTFOUND;
