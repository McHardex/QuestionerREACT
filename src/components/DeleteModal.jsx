import React from 'react';
import SweetAlert from 'react-bootstrap-sweetalert';
import propTypes from 'prop-types';

const DeleteModal = ({ confirmDelete, closeModal }) => (
  <div className="sweet-alert">
    <SweetAlert
      warning
      showCancel
      confirmBtnText="Yes, delete it!"
      confirmBtnBsStyle="danger"
      cancelBtnBsStyle="default"
      title="Are you sure?"
      onConfirm={confirmDelete}
      onCancel={closeModal}
    >
      You will not be able to recover this Meetup!
    </SweetAlert>
  </div>
);

DeleteModal.propTypes = {
  confirmDelete: propTypes.func.isRequired,
  closeModal: propTypes.func.isRequired,
};

export default DeleteModal;
