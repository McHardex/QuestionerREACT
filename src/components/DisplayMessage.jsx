import React from 'react';
import { CSSTransition } from 'react-transition-group';
import propTypes from 'prop-types';

const DisplayMessage = ({
  error, message, onClick, successClass,
}) => (
  <CSSTransition
    in={error}
    timeout={5000}
    classNames="alert"
    unmountOnExit
  >
    <div className={`error-cont ${successClass}`}>
      <p className="error">{message}</p>
      <span className="exit-error" role="presentation" onClick={onClick}>X</span>
    </div>
  </CSSTransition>
);


DisplayMessage.defaultProps = {
  successClass: '',
  message: null,
  error: false,
};

DisplayMessage.propTypes = {
  error: propTypes.bool,
  message: propTypes.string,
  onClick: propTypes.func.isRequired,
  successClass: propTypes.string,
};

export default DisplayMessage;
