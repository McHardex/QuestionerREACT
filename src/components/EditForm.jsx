import React, { Component } from 'react';
import propTypes from 'prop-types';
import { WithContext as ReactTags } from 'react-tag-input';
import Label from './Label';
import '../assets/stylesheets/editForm.css';


export class EditForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      tags: [],
    };
  }

  handleDelete = (i) => {
    const { tags } = this.state;
    this.setState({
      tags: tags.filter((tag, index) => index !== i),
    });
  }

  handleAddition = (tag) => {
    this.setState(state => ({ tags: [...state.tags, tag] }));
  }

  handleDrag = (tag, currPos, newPos) => {
    const { tags: tagsInState } = this.state;
    const tags = [...tagsInState];
    const newTags = tags.slice();

    newTags.splice(currPos, 1);
    newTags.splice(newPos, 0, tag);

    this.setState({ tags: newTags });
  }

  updateMeetup = (e) => {
    e.preventDefault();
    const { tags } = this.state;
    const tagInputs = tags.map(tag => tag.text);
    const data = {
      tags: tagInputs,
    };
    const { target } = e;
    const formData = new FormData(target);
    for (const entry of formData.entries()) {
      const [keys, values] = entry;
      data[keys] = values;
    }

    const { updateMeetup } = this.props;
    updateMeetup(e.target.id, data);
    e.target.reset();
    this.setState({ tags: [] });
  }

  handleTag1 = (e) => {
    e.preventDefault();
    this.setState({ tag1: e.target.value });
    const { tags, tag1 } = this.state;
    tags.push(tag1);
  }

  handleTag2 = (e) => {
    e.preventDefault();
    this.setState({ tag2: e.target.value });
    const { tags, tag2 } = this.state;
    tags.push(tag2);
  }

  handleTag3 = (e) => {
    e.preventDefault();
    this.setState({ tag3: e.target.value });
    const { tags, tag3 } = this.state;
    tags.push(tag3);
  }

  render() {
    const { closeEditModal, singleMeetup } = this.props;
    const { tags } = this.state;

    return (
      <div className="edit-form-cont">
        <form className="edit-form" id={singleMeetup[0] && singleMeetup[0].meetupId} onSubmit={this.updateMeetup}>
          <Label htmlFor="topic">Topic</Label>
          <input
            className="edit-input"
            name="topic"
            type="text"
            defaultValue={singleMeetup[0] && singleMeetup[0].topic}
            required
          />
          <Label htmlFor="date">Date</Label>
          <input
            className="edit-input empty"
            name="happeningOn"
            type="date"
            // defaultValue={convertDate.includes('+') ? removeChar : convertDate}
            required
          />
          <Label htmlFor="location">Location</Label>
          <input
            className="edit-input"
            name="location"
            type="text"
            defaultValue={singleMeetup[0] && singleMeetup[0].location}
            required
          />
          <Label htmlFor="tags">Tags</Label>
          <div className="tags-cont">
            <div className="tags-cont-show">
              <ReactTags
                tags={tags}
                labelField="text"
                placeholder="Add tags"
                handleDelete={this.handleDelete}
                handleAddition={this.handleAddition}
                handleDrag={this.handleDrag}
                maxLength={12}
              />
            </div>
          </div>
          <button type="submit" className="submit-edit">Update Meetup</button>
          <span className="close-edit-tab" onClick={closeEditModal} role="presentation">X</span>
        </form>
      </div>
    );
  }
}

EditForm.defaultProps = {
  singleMeetup: [{}],
};

EditForm.propTypes = {
  closeEditModal: propTypes.func.isRequired,
  updateMeetup: propTypes.func.isRequired,
  singleMeetup: propTypes.arrayOf(propTypes.shape),
};

export default EditForm;
