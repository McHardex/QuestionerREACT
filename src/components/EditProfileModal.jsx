import React from 'react';
import propTypes from 'prop-types';
import Label from './Label.jsx';

const EditProfileModal = ({
  firstname,
  lastname,
  username,
  othername,
  closeModal,
  submitForm,
}) => (
  <div className="profile-details">
    <form className="formCont" onSubmit={submitForm} autoComplete="off">
      <div className="wrap-input">
        <Label className="editProfile-label" htmlFor="Firstname">
          Firstname:
        </Label>
        <input
          type="text"
          className="edit-profile"
          name="firstname"
          defaultValue={firstname}
        />
      </div>
      <div className="wrap-input">
        <Label className="editProfile-label" htmlFor="Lastname">
          Lastname:
        </Label>
        <input
          type="text"
          className="edit-profile"
          name="lastname"
          defaultValue={lastname}
        />
      </div>
      <div className="wrap-input">
        <Label className="editProfile-label" htmlFor="Othername">
          Othername:
        </Label>
        <input
          type="text"
          className="edit-profile"
          name="othername"
          defaultValue={othername}
        />
      </div>
      <div className="wrap-input">
        <Label className="editProfile-label" htmlFor="Username">
          Username:
        </Label>
        <input
          type="text"
          className="edit-profile"
          name="username"
          defaultValue={username}
        />
      </div>
      <button type="submit" className="submit-profile-edit">
        Update Profile
      </button>
    </form>
    <button type="button" className="close-edit-profile" onClick={closeModal}>
      X
    </button>
  </div>
);

EditProfileModal.defaultProps = {
  firstname: null,
  lastname: null,
  username: null,
  othername: null,
};

EditProfileModal.propTypes = {
  firstname: propTypes.string,
  lastname: propTypes.string,
  username: propTypes.string,
  othername: propTypes.string,
  closeModal: propTypes.func.isRequired,
  submitForm: propTypes.func.isRequired,
};
export default EditProfileModal;
