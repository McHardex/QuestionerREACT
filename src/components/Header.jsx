import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import { getCurrentUser } from '../actions/meetupActions';
import '../assets/stylesheets/landingPage.css';
import Logout from './Logout';


class Header extends Component {
  componentDidMount = () => {
    const { getCurrentUser } = this.props;
    getCurrentUser();
  }

  render() {
    const { meetups } = this.props;
    const { user } = meetups;
    const { isAdmin, username } = user;
    return (
      <div>
        {
          isAdmin && username ? (
            <header>
              <div className="questioner-logo">
                <Link to="/meetups"><img src="https://res.cloudinary.com/mchardex/image/upload/v1558174712/logo.png" alt="questioner-logo" /></Link>
                <span>QUESTIONER</span>
              </div>
              <div className="user">
                <i className="fas fa-user-cog" />
                <span id="user">{username}</span>
                <div className="dropdown-content">
                  <Link to="/profile">Profile </Link>
                  <Link to="/meetups">Meetup </Link>
                  <Link to="/admin" id="admin">Admin</Link>
                  <Logout />
                </div>
              </div>
            </header>
          )
            : (
              <header>
                <div className="questioner-logo">
                  <Link to="/meetups"><img src="https://res.cloudinary.com/mchardex/image/upload/v1558174712/logo.png" alt="questioner-logo" /></Link>
                  <span>QUESTIONER</span>
                </div>
                <div className="user">
                  <i className="fas fa-user-cog" />
                  <span id="user">{username}</span>
                  <div className="dropdown-content">
                    <Link to="/profile">Profile </Link>
                    <Link to="/meetups">Meetup </Link>
                    <Logout />
                  </div>
                </div>
              </header>
            )
        }
      </div>
    );
  }
}

Header.propTypes = {
  getCurrentUser: propTypes.func.isRequired,
  meetups: propTypes.shape(propTypes.shape).isRequired,
};

const mapStateToProps = meetups => meetups;

export default connect(mapStateToProps, { getCurrentUser })(Header);
