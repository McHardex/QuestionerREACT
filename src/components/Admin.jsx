import React, { Component } from 'react';
import '../assets/stylesheets/admin.css';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { WithContext as ReactTags } from 'react-tag-input';
import propTypes from 'prop-types';
import Header from './Header';
import Label from './Label';
import { getCurrentUser, getAllMeetups } from '../actions/meetupActions';
import { createMeetup, clearError } from '../actions/adminActions';
import DisplayMessage from './DisplayMessage';
import Loader from './Loader';

class Admin extends Component {
  constructor(props) {
    super(props);

    this.state = {
      searchValue: null,
      length: null,
      meetup: null,
      tags: [],
    };
  }

  handleDelete = (i) => {
    const { tags } = this.state;
    this.setState({
      tags: tags.filter((tag, index) => index !== i),
    });
  }

  handleAddition = (tag) => {
    this.setState(state => ({ tags: [...state.tags, tag] }));
  }

  handleDrag = (tag, currPos, newPos) => {
    const { tags: tagsInState } = this.state;
    const tags = [...tagsInState];
    const newTags = tags.slice();

    newTags.splice(currPos, 1);
    newTags.splice(newPos, 0, tag);

    this.setState({ tags: newTags });
  }

  searchMeetups = (e) => {
    e.preventDefault();
    const searchContent = e.target.value.toLowerCase();
    const { meetups } = this.props;
    const searchResult = meetups.meetups.filter(meetup => (
      meetup.topic.toLowerCase().includes(searchContent)
      || meetup.location.toLowerCase().includes(searchContent)
      || meetup.tags.join(' ').toLowerCase().includes(searchContent)
    ));

    this.setState({
      searchValue: e.target.value,
      length: searchResult.length,
      meetup: searchResult,
    });
  }

  postMeetup = (e) => {
    e.preventDefault();
    const { tags } = this.state;
    const tagInputs = tags.map(tag => tag.text);
    const data = {
      tags: tagInputs,
    };
    const { target } = e;
    const formData = new FormData(target);
    for (const entry of formData.entries()) {
      const [keys, values] = entry;
      data[keys] = values;
    }

    const { createMeetup, getAllMeetups } = this.props;
    createMeetup(data, () => {
      getAllMeetups();
      target.reset();
      this.setState({ tags: [] });
    });
  }

  clearError = () => {
    const { clearError } = this.props;
    clearError();
  }

  componentDidMount = () => {
    const {
      getCurrentUser, getAllMeetups,
    } = this.props;
    getCurrentUser();
    getAllMeetups();
  }

  componentWillReceiveProps = (nextProps) => {
    this.setState({
      meetup: nextProps.meetups.meetups,
    });
  }

  render() {
    const { meetups, admin } = this.props;
    const { message, postMeetupError, isLoading } = admin;
    const { user } = meetups;
    const { isAdmin, username } = user;

    const {
      meetup, length, searchValue, tags,
    } = this.state;

    return (
      <div className="admin-cont">
        <DisplayMessage
          error={postMeetupError}
          message={message}
          onClick={this.clearError}
        />
        <Header role={isAdmin} username={username} />
        <div className="cont">
          <div className="create-meetup-bk">
            <div className="banner-text">
              <h1>Create a new Meetup</h1>
              <p>Enrich lives through open mindset and constructive discussions</p>
            </div>
          </div>
          <div className="cte-meetup-cont">
            <form className="create-meetup" id="create-meetup" onSubmit={this.postMeetup}>
              <Label htmlFor="topic">Topic</Label>
              <input className="form-input" name="topic" type="text" required />
              <Label htmlFor="date">Date</Label>
              <input className="form-input empty" name="happeningOn" type="date" required />
              <Label htmlFor="location">Location</Label>
              <input className="form-input" name="location" type="text" required />
              <Label htmlFor="tags">Tags</Label>
              <div className="tags-cont-show">
                <ReactTags
                  tags={tags}
                  labelField="text"
                  handleDelete={this.handleDelete}
                  handleAddition={this.handleAddition}
                  handleDrag={this.handleDrag}
                  maxLength={12}
                />
              </div>
              <button type="submit" className="submit-form">Create Meetup</button>
            </form>
          </div>
          <hr />
          <div className="search-cont">
            <input
              type="text"
              name="search"
              autoComplete="on"
              placeholder="Search meetup by title, location or tags"
              className="search-meetup"
              onChange={this.searchMeetups}
              tabIndex="-2"
            />
          </div>
          <div id="overlay" />
          <div className="meetups" id="meetups" />
        </div>
        {length === 0
          ? (
            <div className="meetups">
              <p className="not-found">
                Sorry, we could not find any meetups matching
                {' '}
                {`"${searchValue}"`}
              </p>
            </div>
          )
          : (
            <div className="meetups" id="meetups">
              {
                meetup && meetup.map(meetup => (
                  <Link to={`/meetups/${meetup.id}`} className="link" key={meetup.id}>
                    <div
                      className="meetup-cont"
                      role="presentation"
                      key={meetup.id}
                    >
                      <div className="meetup-text">
                        <p className="date">{new Date(meetup.happeningon).toDateString()}</p>
                        <h3 id={meetup.id} className="meetup-topic">{meetup.topic}</h3>
                        <p className="loctn">{meetup.location}</p>
                        <div className="tags-cont">
                          {meetup.tags.map(tag => (
                            <span key={tag}>{tag}</span>
                          ))}
                        </div>
                      </div>
                      <i className="fas fa-trash" title="delete" id={meetup.id} />
                      <i className="far fa-edit" title="edit" id={meetup.id} />
                    </div>
                  </Link>
                ))
              }
            </div>
          )
        }
        {isLoading && <Loader />}
      </div>
    );
  }
}

Admin.propTypes = {
  getCurrentUser: propTypes.func.isRequired,
  getAllMeetups: propTypes.func.isRequired,
  createMeetup: propTypes.func.isRequired,
  clearError: propTypes.func.isRequired,
  meetups: propTypes.objectOf(propTypes.shape).isRequired,
  admin: propTypes.objectOf(propTypes.shape).isRequired,
};

const mapStateToProps = ({ admin, meetups }) => ({ admin, meetups });

export default connect(mapStateToProps, {
  getCurrentUser,
  getAllMeetups,
  createMeetup,
  clearError,
})(Admin);
