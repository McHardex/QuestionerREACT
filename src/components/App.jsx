/* eslint-disable import/extensions */
import React from 'react';
import { BrowserRouter as Router, Route, Switch, Redirect } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../lib/store';
import LandingPage from './LandingPage';
import Signup from './Signup';
import Login from './Login';
import Admin from './Admin';
import Meetup from './Meetup';
import MeetupDetails from './MeetupDetails';
import Profile from './Profile';
import NOTFOUND from './404';

const App = () => (
  <div>
    <Provider store={store}>
      <Router>
        <div>
          <Switch>
            <Route exact path="/" component={LandingPage} />
            <Route exact path="/signup" component={Signup} />
            <Route exact path="/login" component={Login} />
            <Route exact path="/meetups" component={Meetup} />
            <Route exact path="/admin" component={Admin} />
            <Route exact path="/meetups/:id" component={MeetupDetails} />
            <Route exact path="/profile" component={Profile} />

            <Route path="/signup" component={Signup} />
            <Route path="/login" component={Login} />
            <Route path="/meetups" component={Meetup} />
            <Route path="/admin" component={Admin} />
            <Route path="/not-found" component={NOTFOUND} />
            <Redirect to="/not-found" />
          </Switch>
        </div>
      </Router>
    </Provider>
  </div>
);

export default App;
