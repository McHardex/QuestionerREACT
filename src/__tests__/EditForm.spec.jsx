import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { EditForm } from '../components/EditForm';


const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({});

describe('EditForm component', () => {
  const props = {
    closeEditModal: jest.fn(),
    updateMeetup: jest.fn(),
    singleMeetup: [
      { happeningon: '2019-04-16T14:49:26.699Z' },
      { happeningon: '2019-04-16T14:49:26.699Z' },
    ],
  };

  it('snapshot', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <EditForm {...props} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleAddition method', () => {
    const wrapper = shallow(<EditForm {...props} />);
    expect(wrapper.instance().handleAddition());
  });

  it('should call handleDelete method', () => {
    const wrapper = shallow(<EditForm {...props} />);
    expect(wrapper.instance().handleDelete());
  });

  it('should call handleDrag method', () => {
    const wrapper = shallow(<EditForm {...props} />);
    expect(wrapper.instance().handleDrag());
  });

  it('should call handleTag1 method', () => {
    const wrapper = shallow(<EditForm {...props} />);
    const e = { preventDefault: jest.fn(), target: { value: 'val' } };
    expect(wrapper.instance().handleTag1(e));
  });

  it('should call handleTag2 method', () => {
    const wrapper = shallow(<EditForm {...props} />);
    const e = { preventDefault: jest.fn(), target: { value: 'val' } };
    expect(wrapper.instance().handleTag2(e));
  });

  it('should call handleTag3 method', () => {
    const wrapper = shallow(<EditForm {...props} />);
    const e = { preventDefault: jest.fn(), target: { value: 'val' } };
    expect(wrapper.instance().handleTag3(e));
  });
});
