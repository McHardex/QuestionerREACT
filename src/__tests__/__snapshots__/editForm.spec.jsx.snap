// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditForm component snapshot 1`] = `ShallowWrapper {}`;
